packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -0,0 +1,562 @@
+      childParentData.offset = new Offset(crossPosition, currentOffset);
+      currentOffset += _itemExtent;
+      child = childAfter(child);
+    }
+
+    offset.applyViewportDimension(_viewportExtent);
+    offset.applyContentDimensions(_minScrollExtent, _maxScrollExtent);
+  }
+
+  bool _shouldClipAtCurrentOffset() {
+    final double highestUntransformedPaintY =
+        _getUntransformedPaintingCoordinateY(0.0);
+    return highestUntransformedPaintY < 0.0
+        || size.height < highestUntransformedPaintY + _maxScrollExtent + _itemExtent;
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (childCount > 0) {
+      if (_clipToSize && _shouldClipAtCurrentOffset()) {
+        context.pushClipRect(
+          needsCompositing,
+          offset,
+          Offset.zero & size,
+          _paintVisibleChildren,
+        );
+      } else {
+        _paintVisibleChildren(context, offset);
+      }
+    }
+  }
+
+  /// Visits all the children until one is partially visible in the viewport.
+  RenderBox _getFirstVisibleChild() {
+    assert(childCount > 0);
+    final double firstVisibleLayoutOffset = _firstVisibleLayoutOffset;
+
+    RenderBox child = firstChild;
+    _ListWheelParentData childParentData = child.parentData;
+
+    while (childParentData != null
+        && childParentData.offset.dy <= firstVisibleLayoutOffset) {
+      child = childAfter(child);
+      childParentData = child?.parentData;
+    }
+
+    return child;
+  }
+
+  /// Paints all children visible in the current viewport.
+  void _paintVisibleChildren(PaintingContext context, Offset offset) {
+    assert(childCount > 0);
+    final double lastVisibleLayoutOffset = _lastVisibleLayoutOffset;
+
+    RenderBox childToPaint = _getFirstVisibleChild();
+    _ListWheelParentData childParentData = childToPaint?.parentData;
+
+    while (childParentData != null
+        && childParentData.offset.dy < lastVisibleLayoutOffset) {
+      _paintTransformedChild(childToPaint, context, offset, childParentData.offset);
+      childToPaint = childAfter(childToPaint);
+      childParentData = childToPaint?.parentData;
+    }
+  }
+
+  /// Takes in a child with a **scrollable layout offset** and paints it in the
+  /// **transformed cylindrical space viewport painting coordinates**.
+  void _paintTransformedChild(
+    RenderBox child,
+    PaintingContext context,
+    Offset offset,
+    Offset layoutOffset,
+  ) {
+    final Offset untransformedPaintingCoordinates = offset
+        + new Offset(
+            layoutOffset.dx,
+            _getUntransformedPaintingCoordinateY(layoutOffset.dy)
+        );
+
+    // Get child's center as a fraction of the viewport's height.
+    final double fractionalY =
+        (untransformedPaintingCoordinates.dy + _itemExtent / 2.0) / size.height;
+    final double angle = -(fractionalY - 0.5) * 2.0 * _maxVisibleRadian;
+    // Don't paint the backside of the cylinder when
+    // renderChildrenOutsideViewport is true.
```

Comment:
reading this it's not clear that renderChildrenOutsideViewport is being inspected here. How does it contribute to the angle? This should be explained in the comment.

Reply:
Added some comments. Only children within suitable angle get here via math in _paintVisibleChildren/firstVisibleLayoutOffset. The math is complicated with renderChildrenOutsideViewport so it's filtered here.